specs/CriticalRestartRequired.md

```diff
@@ -0,0 +1,105 @@
+Critical Update Available
+===
+
+# Background
+As WebView2 developers, we often have to author ECS (experimentation and configuration service - 
+used to perform A/B testing or remotely disable features) configurations to toggle feature flags. 
+However, once these payloads are received, there is no way to restart the WebView2 and apply the
+payload. The purpose of this API is to detect such critical payloads and inform the end developer
+so they may restart their app or their WebView2 or other appropriate action.
```

Comment:
Does "restart" mean e.g. [AppInstance.Restart](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.Windows.AppLifecycle.AppInstance.Restart)? Would help to show this in the example.

Is there a way to just restart the WV2 though?

Reply:
I'm also curious as to how much "restarting" is needed. For example, say there are two apps using WebView2 (call then X and Y). A critical restart is requested. X shuts down its WebView2 objects but Y does not. When X recreates its WebView2, it will still use the same WebView2 process pool as Y, which didn't restart. What happens now? Did the features change only for X and not Y? Did it change for both? For neither? Does X have to wait for Y to restart its webview2 before X can recreate its own webview2?

If X waits for BrowserProcessExited before recreating its webview2, could it be left stranded because Y never exits, resulting in an accidental denial of service attack against X? (Because X will be displaying a spinning "Please wait while we apply a critical update to your browser" forever?)

Reply:
If multiple apps are coordinating to share a user data folder and browser process then they must also coordinate to shutdown their webview2s at the same time. Otherwise like you say the same browser process will be running and it will still have the old configuration. 

@dianaqu please ensure we mention this in the ref docs for the event.

Reply:
The section below this part already mentioned cases for sharing UDF